### ISSUE TITLE
lstm_seq2seq.py not working correctly, gives same output no matter the input

### ISSUE THREAD

**Issue by eddiemay** [keras-team/keras-io#1906]
### Issue Type

Bug

### Source

source

### Keras Version

Keras 3.3.3

### Custom Code

No

### OS Platform and Distribution

macOS 14.5

### Python version

3.11.9

### GPU model and memory

Apple M1 

### Current Behavior?

model.predict will give the same exact result no matter the input. I have done simple examples to just echo back the input where input and output are the same and still can't get it to have an accuracy above 76% then when model is verified at the end it will output the same result no matter the input. I finally just downloaded the tutorial code without modification and it too gives the same output no matter the input.

### Standalone code to reproduce the issue or tutorial link

```shell
https://github.com/keras-team/keras-io/blob/master/examples/nlp/lstm_seq2seq.py
```

### Relevant log output

```shell
-
Input sentence: Go.
Decoded sentence: Je suis prêt.

-
Input sentence: Go.
Decoded sentence: Je suis prêt.

-
Input sentence: Go.
Decoded sentence: Je suis prêt.

-
Input sentence: Go.
Decoded sentence: Je suis prêt.

-
Input sentence: Hi.
Decoded sentence: Je suis prêt.

-
Input sentence: Hi.
Decoded sentence: Je suis prêt.

-
Input sentence: Run!
Decoded sentence: Je suis prêt.

-
Input sentence: Run!
Decoded sentence: Je suis prêt.

-
Input sentence: Run!
Decoded sentence: Je suis prêt.

-
Input sentence: Run!
Decoded sentence: Je suis prêt.

-
Input sentence: Run!
Decoded sentence: Je suis prêt.

-
Input sentence: Run!
Decoded sentence: Je suis prêt.

-
Input sentence: Run!
Decoded sentence: Je suis prêt.

-
Input sentence: Run!
Decoded sentence: Je suis prêt.

-
Input sentence: Run.
Decoded sentence: Je suis prêt.

-
Input sentence: Run.
Decoded sentence: Je suis prêt.

-
Input sentence: Run.
Decoded sentence: Je suis prêt.

-
Input sentence: Run.
Decoded sentence: Je suis prêt.

-
Input sentence: Run.
Decoded sentence: Je suis prêt.

-
Input sentence: Run.
Decoded sentence: Je suis prêt.
```

**Comment by eddiemay** [comment]
I have tested it with the latest version of Keras (3.4.1) and it seems to preform better. The output is not the same as the code lab but it is at least giving different outputs for different inputs. Keras 3.4.1 is working good for the actual project I am working on with this code which is transforming Messoritic Hebrew Text into Dead Sea Scroll Hebrew i.e. (transform אֱלֹהֵ֖ינוּ to אלוהינו)

**Comment by sonali-kumari1** [comment]
Hi @eddiemay - 
Thanks for reporting this issue!

I have tested [lstm_seq2seq.py](https://github.com/keras-team/keras-io/blob/master/examples/nlp/lstm_seq2seq.py) example with the latest version of keras(3.11.0) in this [gist ](https://colab.sandbox.google.com/gist/sonali-kumari1/2a4492f6d75e0e57b8daf75656834060/lstm_seq2seq.ipynb)file. The training accuracy improves from ~73% to ~95%, and validation accuracy from ~71% to ~87%, confirming that the model trains properly. While the decoded outputs vary with different inputs, and don't match with the official docs, this is likely due to random initialization of weights, shuffling of training data, and other nondeterministic factors that can influence training.
